src/prefect/tasks/postgres/postgres.py

```diff
@@ -0,0 +1,200 @@
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+import psycopg2 as pg
+
+
+class PostgresExecute(Task):
+    """
+        Task for executing a query against a Postgres database.
+
+        Args:
+            - db_name (str): name of Postgres database
+            - user (str): user name used to authenticate
+            - password (str): password used to authenticate
+            - host (str): database host address
+            - port (int, optional): port used to connect to Postgres database, defaults to 5432 if not provided
+            - query (str, optional): query to execute against database
+            - data (tuple, optional): values to use in query, must be specified using placeholder is query string
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+            - **kwargs (dict, optional): additional keyword arguments to pass to the
+                Task constructor
+        """
+
+    def __init__(
+        self,
+        db_name: str,
+        user: str,
+        password: str,
+        host: str,
+        port: int = 5432,
+        query: str = "",
+        data: tuple = None,
+        commit: bool = False,
+        **kwargs
+    ):
+        self.db_name = db_name
+        self.user = user
+        self.password = password
+        self.host = host
+        self.port = port
+        self.query = query
+        self.data = data
+        self.commit = commit
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("query", "data", "commit")
+    def run(self, query: str = "", data: tuple = None, commit: bool = False):
+        """
+        Task run method. Executes a query against Postgres database.
+
+        Args:
+            - query (str, optional): query to execute against database
+            - data (tuple, optional): values to use in query, must be specified using
+                placeholder is query string
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+
+        Returns:
+            - None
+        """
+        if query == "":
```

Comment:
Let's default the `query` to None and change this to `if not query` so it'll catch the cases where it's None and ""